### ISSUE TITLE
Job.fromId String Parameter Contradicts Method Comment/Implementation

### ISSUE THREAD

**Issue by bobdercole** [taskforcesh/bullmq#134]
The `jobId` parameter in the following method accepts `string`. However, the comment in the method states that it could be `undefined`. The conditional then proceeds to check for a truthy value.
https://github.com/taskforcesh/bullmq/blob/36726bfb01430af8ec606f36423fc187e4a06fb4/src/classes/job.ts#L138-L145

This was brought up while trying to fix the return type of the method in #133. I believe we came to the conclusion to remove the conditional check. We would also need to review any code that consumes `Job.fromId` to ensure `undefined` isn't being passed from anywhere.

These could be easily found by switching `strictNullChecks` to true. I believe this was brought up in GH-112.

https://github.com/taskforcesh/bullmq/blob/36726bfb01430af8ec606f36423fc187e4a06fb4/tsconfig.json#L16

An alternative is to simply change the parameter to `jobId?: string`. However, I feel like it's weird to have an optional ID parameter on a method named fromId.

I could make a PR once a merge decision is made for PR #133, since the code is sort of intertwined.

References:
https://github.com/taskforcesh/bullmq/pull/133#discussion_r377365170
https://github.com/taskforcesh/bullmq/issues/130#issuecomment-584286855

**Comment by manast** [comment]
I agree that a method that does nothing if you do not provide the second parameter is very awkward. These are remains from older code, thats why it looks like this. So best is to keep it mandatory, but as you write we need to make sure that all the calls to this method always pass a defined jobId.
